src/pages/ReferralPage.tsx

```diff
@@ -0,0 +1,26 @@
+import { useEffect } from 'react';
+
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { DialogTypes } from '@/constants/dialogs';
+import { DEFAULT_TRADE_ROUTE } from '@/constants/routes';
+
+import { useAppDispatch } from '@/state/appTypes';
+import { openDialog } from '@/state/dialogs';
+
+const ReferralPage = () => {
+  const dispatch = useAppDispatch();
+  const { refCode } = useParams<{ refCode: string }>();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (refCode) {
+      dispatch(openDialog(DialogTypes.Referral({ refCode })));
+      navigate(DEFAULT_TRADE_ROUTE, { replace: true });
+    }
+  }, [refCode, navigate, dispatch]);
+
```

Comment:
is this supposed to always return null? if someone lands here without a refCode will they just see a blank page?

Reply:
In theory it should never land here. since if refCode null or empty it should fallback to this route:
`<Route index element={<Navigate to={DEFAULT_TRADE_ROUTE} replace />} />`

I did test routes without refCode, and it seems to be ok. but maybe there is a better way to do this

Reply:
hmm got it. i'm a little surprised that dispatching the dialog open and navigating behaves properly.

can we instead add this functionality to the trade page directly? where if there's a ref code we open a dialog? 🤔 if that's significantly more difficult then this is probablyyy fine, but may warrant a comment what's going on here since it took a bit to understand

Reply:
I guess I don't the referral dialog think this should be a trade page specific behavior, especially if we A/B test what the landing page should be. Also we would still need to create a referral route to capture the refCode before redirecting

Reply:
discussed [on slack](https://dydx-team.slack.com/archives/C042V24U87Q/p1726244192767569) - changing this to use a search param instead of a path. this component no longer needed (right?)

Reply:
should be good now.